### DISCUSSION TITLE
LAB 02.03 - PANDAS -task 1

### DISCUSSION THREAD

**Discussion by diegosaaval** [rramosp/ai4eng.v1#134] [UDEA Students]
Buenas tardes, tengo una duda sobre este punto sobre la autocorrección y es que cuando califico la tarea me dice que mi nota es 0, esto sucede debido a que la lista que se espera tiene un orden diferente, pero los elementos de la lista son los mismos

Agradezco de antemano si me colaboran por favor
![1](https://user-images.githubusercontent.com/106838105/171936678-e143247a-951e-4ac8-9f72-b757646fa8e5.png)
![2](https://user-images.githubusercontent.com/106838105/171936686-38f2d01e-b57d-4fae-80e9-519002526eff.png)

**Comment by Felipe-RA** [comment]
Saludos Diego

Efectivamente, se espera que tu output sea en el orden que los `itemid` aparecen en el dataframe

**Reply by diegosaaval** [reply]
Gracias por la respuesta, entiendo, es que fue de la única manera que me entregaba los números

Agradecería mucho si por favor me ayudaran con el código para que los arroje en ese orden, intenté sumando, colocando OR, AND y nada funcionaba

En unos casos me cogía los (>100) y los (>100 y >10) pero no me cogía SOLO los (> 10)

En otro caso si me cogía los (>10) y los (>100 y >10) pero no me cogía SOLO los (>100)

Sé que es un poco confuso, espero que me haya hecho entender, quedo atento

**Reply by Felipe-RA** [reply]
Por supuesto,

Veo que ya estas familiarizado con la aplicación de condicionales a los dataframes (cuando haces `df["margin"] > 10`)
A esto también puedes agregarle directamente un condicional como **OR** `|` y **AND** `&`

Entonces, puedes juntar las dos condiciones que tienes y aplicarlas en una sola operación sobre el dataframe completo: 

`df[df["price"]>100 | df["margin"]>10]`

Te recomiendo usar lo siguiente para simplificar tu código:
`result.index.tolist()`

Así no tienes que llamar a df.index cada vez, sólo tienes que pasar el index a lista al final. Ayuda con la legibilidad de tu código

Cuentame si necesitas algo más luego!

**Reply by diegosaaval** [reply]
Te agradezco demasiado la ayuda, seguí todas las sugerencias y me funcionó de maravilla, me seguía dando otro error mínimo, pero se solucionó agregando unos paréntesis

**Comment by Felipe-RA** [comment]
Hola @DiegoSaav 

Fijate en la diferencia entre el output de `d["price"]>100`: 

````
itemid
76401    False
58675    False
33334     True
94927     True
46326     True
57242    False
46289     True
79879    False
89757    False
67541    False
Name: price, dtype: bool
````

Y el de `d[d["price"]>100]`:

```
         price  category  margin
itemid                          
33334   104.23       0.0   11.05
94927   101.91       0.0   10.56
46326   117.63       2.0   10.76
46289   105.20       1.0   10.28
```

Con el primero, estás creando un DataFrame con valores booleanos según si cumplen la condición o no.

Mientras que en el segundo le estás entregando al dataframe original este dataframe con dtype: bool para que filtre los valores según el index asociado al booleano.

Entonces, el error raro que te sale es por la imposibilidad del grader para evaluar un dataframe con dtype bool, pues este espera un dataframe con valores numéricos.

**Recuerda siempre ante la duda probar por fuera  de la definición de la función cada parte de tu código**

**Reply by diegosaaval** [reply]
Excelente la argumentación Felipe lo tendré muy presente, muchas gracias
